File: Entidad/Factura.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace Entidad
{
    public class Factura
    {
        public Factura(List<DetalleFactura> detallesFactura)
        {
            DetallesFactura = detallesFactura;
        }

        public Factura()
        {

        }

        [Key]
        public int FacturaId { get; set; }
        public DateTime Fecha { get; set; }
        [Display(Name = "CantidadServicios")]
        public int CantidadServicios 
        {
            get{
                return DetallesFactura.Count();
            }
        }
        
        [Display(Name = "SubTotal")]
        public double SubTotal 
        {
            get{
                return DetallesFactura.Sum(s => s.SubTotal);
            }
        }
        [Display(Name = "ValorIva")]
        public double ValorIva 
        {
            get{
                return  DetallesFactura.Sum(s => s.ValorIva); 
            }
        }
        [Display(Name = "ValorDescuento")]
        public double ValorDescuento
        {
            get{
                return DetallesFactura.Sum(s => s.ValorDescuento); 
            }
        }
        [Display(Name = "Total")]
        public double Total 
        {
            get{
                return SubTotal  + ValorIva - ValorDescuento;
            }
        }
        [Required]
        public  string Identificacion { get; set; }
        public virtual List<DetalleFactura> DetallesFactura { get; set; }
    }
}

File: veterinariadotnet/ClientApp/src/app/veterinaria/Modelos/Factura/factura.ts
import { Cliente } from "../../Componentes/models/cliente";
import { DetalleFactura } from "../Detalle Factura/detalle-factura";

export class Factura {
  facturaId :number;
  fecha : Date = new Date();
  cantidad : number;
  subTotal : number;
  total : number;
  valorDescuento : number;
  valorIva : number;
  detallesFactura : DetalleFactura [] = [];
  identificacion : string;

   RealizarCalculos()
  {
    this.CalcularDescuento();
    this.CalcularSubtotal();
    this.CalcularIva();
    this.CalcularTotal();
  }

  CalcularTotal (){
    this.total = 0;
    this.detallesFactura.forEach( detalle => {
      this.total = this.total + detalle.total;
    });
    this.total = this.total - this.CalcularDescuento() + this.CalcularIva();
    return this.total;
  }
  CalcularSubtotal (){
    this.subTotal = 0;
    this.detallesFactura.forEach( detalle => {
      this.subTotal = this.subTotal + detalle.subTotal;
    });
    return this.total;
  }
  CalcularIva (){
    this.valorIva = 0;
    this.detallesFactura.forEach( detalle => {
      this.valorIva = this.valorIva + detalle.valorIva;
    });
    return this.valorIva;
  }
  CalcularDescuento (){
    this.valorDescuento = 0;
    this.detallesFactura.forEach( detalle => {
      this.valorDescuento = this.valorDescuento + detalle.valorDescuento;
    });
    return this.valorDescuento;
  }
}
